src/main/java/attendance/model/attendance/log/AttendanceLogs.java

```diff
@@ -0,0 +1,151 @@
+            final LocalDate from,
+            final LocalDate to,
+            final CampusOperationPolicy campusOperationPolicy
+    ) {
+
+        return new AttendanceLogs(Stream.concat(
+                        values.stream(),
+                        getAbsenceAttendanceLogs(from, to, campusOperationPolicy).stream()
+                )
+                .sorted(Comparator.comparing(AttendanceLog::getDate))
+                .toList());
+    }
+
+    private List<AttendanceLog> getAbsenceAttendanceLogs(
+            final LocalDate from,
+            final LocalDate to,
+            final CampusOperationPolicy campusOperationPolicy
+    ) {
+
+        return from.datesUntil(to)
+                .filter(campusOperationPolicy::isOpenDate)
+                .filter(date -> !containsDate(date))
+                .map(date -> AttendanceLog.fromAbsenceDate(date, campusOperationPolicy))
+                .toList();
+    }
+
+    private boolean containsDate(final LocalDate date) {
+        return values.stream().anyMatch(attendanceLog -> attendanceLog.isSameDate(date));
+    }
+
+    public Map<AttendanceStatus, Integer> calculateAttendanceStatusStatistics() {
+        return AttendanceStatus.getStatistics(getAllAttendanceStatuses());
+    }
+
+    public List<AttendanceStatus> getAllAttendanceStatuses() {
+        return values.stream()
+                .map(AttendanceLog::getAttendanceStatus)
+                .toList();
+    }
+
+    public void add(final AttendanceLog attendanceLog) {
+        values.add(attendanceLog);
+    }
+
+    public int getPolicyAppliedAbsenceCount() {
+        return getAbsenceCount() + getLateCount() / 3;
+    }
+
+    public int getAbsenceCount() {
+        return Math.toIntExact(
+                values.stream()
+                        .filter(this::isAbsence)
+                        .count()
+        );
+    }
+
+    public int getPolicyAppliedLateCount() {
+        return getLateCount() % 3;
+    }
+
+    public int getLateCount() {
+        return Math.toIntExact(
+                values.stream()
+                        .filter(this::isLate)
+                        .count()
+        );
+    }
+
+    private boolean isLate(AttendanceLog attendanceLog) {
+        return attendanceLog.getAttendanceStatus() == AttendanceStatus.LATE;
+    }
+
+    private boolean isAbsence(AttendanceLog attendanceLog) {
+        return attendanceLog.getAttendanceStatus() == AttendanceStatus.ABSENCE;
+    }
+
+    public void update(final AttendanceLog from, final AttendanceLog to) {
+        final Optional<AttendanceLog> targetAttendanceLog = values.stream()
+                .filter(log -> log.isSameDate(from.getDate()))
+                .findFirst();
+
+        if (targetAttendanceLog.isPresent()) {
+            values.remove(targetAttendanceLog.get());
+            values.add(to);
+            return;
+        }
+
+        throw new IllegalArgumentException("해당 날짜의 출석 기록이 존재하지 않습니다.");
+    }
```

Comment:
```suggestion
    public void update(final AttendanceLog from, final AttendanceLog to) {
        final AttendanceLog targetAttendanceLog = values.stream()
                .filter(log -> log.isSameDate(from.getDate()))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("해당 날짜의 출석 기록이 존재하지 않습니다."));
        
        values.remove(targetAttendanceLog);
        values.add(to);
    }
```
targetAttendanceLog를 Optional로 반환받지 않아도 로직상 문제가 없을 것 같은데 어떤가요? 😄

Reply:
맞습니다 ㅎㅎ 성숙하지 못한 코드네요

Reply:
아닙니당 이런건 혼자 보면 잘 안보이더라구요